lib/telepoison.ex

```diff
@@ -45,18 +45,39 @@ defmodule Telepoison do
         {"http.url", request.url}
       ] ++ Keyword.get(opts, :ot_attributes, [])
 
-    new_ctx = Tracer.start_span(span_name, %{kind: :client, attributes: attributes})
-    Tracer.set_current_span(new_ctx)
+    request_ctx = Tracer.start_span(span_name, %{kind: :client, attributes: attributes})
+    Tracer.set_current_span(request_ctx)
 
-    super(request)
+    result = super(request)
+
+    if Tracer.current_span_ctx() == request_ctx do
+      case result do
+        {:error, %{reason: reason}} ->
+          Tracer.set_status(:error, inspect(reason))
+          end_span()
```

Comment:
Is this needed or is this a bonus? (since we didn't have it before I assume it worked without having to end the span explicitly)

Reply:
It's needed because in the case of nxdomain, `process_response_status_code` isn't called so the span wouldn't be closed.